packages/terra-menu/src/_MenuContent.jsx

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import List from 'terra-list';
+import IconLeft from 'terra-icon/lib/icon/IconLeft';
+import ContentContainer from 'terra-content-container';
+import IconClose from 'terra-icon/lib/icon/IconClose';
+import Arrange from 'terra-arrange';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import MenuItemGroup from './MenuItemGroup';
+import styles from './Menu.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Title the should be displayed in header
+   */
+  title: PropTypes.string,
+  /**
+   * Callback function for when back button is clicked
+   */
+  onRequestBack: PropTypes.func,
+  /**
+   * Callback function for when close button is clicked
+   */
+  onRequestClose: PropTypes.func,
+  /**
+   * Callback function that takes the content to be displayed next and is called when an item with nested content is clicked
+   */
+  onRequestNext: PropTypes.func.isRequired,
+  /**
+   * Menu Items/Menu Groups/Menu Dividers to be displayed
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Index within the Menu Stack
+   */
+  index: PropTypes.number.isRequired,
+};
+
+const defaultProps = {
+  children: [],
+  title: '',
+};
+
+const childContextTypes = {
+  isSelectableMenu: PropTypes.bool,
+};
+
+class MenuContent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.wrapOnClick = this.wrapOnClick.bind(this);
+    this.buildHeader = this.buildHeader.bind(this);
+    this.isSelectable = this.isSelectable.bind(this);
+  }
+
+  getChildContext() {
+    return { isSelectableMenu: this.isSelectable() };
+  }
+
+  isSelectable() {
+    for (let i = 0; i < this.props.children.length; i += 1) {
+      const child = this.props.children[i];
+      if (child.type === <MenuItemGroup />.type || child.props.isSelectable) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  wrapOnClick(item) {
+    const onClick = item.props.onClick;
+    return (event) => {
+      this.props.onRequestNext(item);
+
+      if (onClick) {
+        onClick(event);
+      }
+    };
+  }
+
+  buildHeader() {
```

Comment:
Is this the same function that builds subheader menus? I'm not sure the tabIndex='0' is applying correctly for icons